packages/react-components/react-charts-preview/library/src/components/HorizontalBarChart/HorizontalBarChart.tsx

```diff
@@ -0,0 +1,394 @@
+    const MARGIN_WIDTH_IN_PX = 3;
+    if (svgWidth) {
+      const currentBarSpacing = (MARGIN_WIDTH_IN_PX / svgWidth) * 100;
+      setBarSpacingInPercent(currentBarSpacing);
+    }
+  }, []);
+
+  const { data } = props;
+  _adjustProps();
+  const classes = useHorizontalBarChartStyles_unstable(props);
+  const focusAttributes = useFocusableGroup();
+
+  let datapoint: number | undefined = 0;
+  return !_isChartEmpty() ? (
+    <div className={classes.root} onMouseLeave={_handleChartMouseLeave}>
+      {data!.map((points: IChartProps, index: number) => {
+        if (points.chartData && points.chartData![0] && points.chartData![0].horizontalBarChartdata!.x) {
+          datapoint = points.chartData![0].horizontalBarChartdata!.x;
+        } else {
+          datapoint = 0;
+        }
+        points.chartData![1] = {
+          legend: '',
+          horizontalBarChartdata: {
+            x: points.chartData![0].horizontalBarChartdata!.y - datapoint!,
+            y: points.chartData![0].horizontalBarChartdata!.y,
+          },
+          color: tokens.colorBackgroundOverlay,
+        };
+
+        // Hide right side text of chart title for absolute-scale variant
+        const chartDataText =
+          props.variant === HorizontalBarChartVariant.AbsoluteScale ? null : _getChartDataText(points!);
+        const bars = _createBars(points!);
+        const keyVal = _uniqLineText + '_' + index;
+        // ToDo - Showtriangle property is per data series. How to account for it in the new stylesheet
+        /*         const classes = useHorizontalBarChartStyles_unstable(props.styles!, {
+          width: props.width,
+          showTriangle: !!points!.chartData![0].data,
+          variant: props.variant,
+        }); */
+
+        return (
+          <div key={index}>
+            <div className={classes.items} {...focusAttributes}>
+              <div className={classes.chartTitle}>
+                {points!.chartTitle && (
+                  <FocusableTooltipText
+                    className={classes.chartTitleLeft}
+                    content={points!.chartTitle}
+                    accessibilityData={points!.chartTitleAccessibilityData}
+                  />
+                )}
+                {chartDataText}
+              </div>
+              {points!.chartData![0].data && _createBenchmark(points!)}
+              <svg ref={barChartSvgRef} className={classes.chart} aria-label={points!.chartTitle}>
+                <g
+                  id={keyVal}
+                  key={keyVal}
+                  ref={(e: SVGGElement) => {
+                    _refCallback(e, points!.chartData![0].legend);
+                  }}
+                  // NOTE: points.chartData![0] contains current data value
+                  onClick={() => {
+                    const p = points!.chartData![0];
+                    if (p && p.onClick) {
+                      p.onClick();
+                    }
+                  }}
```

Comment:
should this be wrapped in a useCallback?

Reply:
Such optimizations will be taken up in following iterations. Tracked in https://github.com/microsoft/fluentui/issues/33042